packages/grafana-runtime/src/services/SidecarService_EXPERIMENTAL.ts

```diff
@@ -20,24 +22,52 @@ export class SidecarService {
   }
 
   get activePluginId() {
+    if (!this.assertFeatureEnabled()) {
+      return EMPTY;
+    }
     return this._activePluginId.asObservable();
   }
 
-  openApp(pluginId: string) {
+  get initialContext() {
     if (!this.assertFeatureEnabled()) {
-      return false;
+      return EMPTY;
     }
+    return this._initialContext.asObservable();
+  }
 
-    return this._activePluginId.next(pluginId);
+  // Get the current value of the subject, this is needed if we want the value immediately. For example if used in
+  // hook in react with useObservable first render would return undefined even if the behaviourSubject has some
+  // value which will be emitted in the next tick and thus next rerender.
+  get initialContextCurrent() {
```

Comment:
@leventebalogh maybe you would know if there is a way around this. Basically how you would not use it in some app is:
```
const sidecarContext = useSidecar_EXPERIMENTAL();
getInitialFilters(sidecarContext.initialContext)
```
The problem is because of the nature of the observable and the `useObservable` the first value of `initialContext` is undefined because the `useObservable` have to wait for the first emit which is async. We can get the current value like this and then use that as a default for first render. So just checking if you know some easier way to do it or not cause I didn't find out a better way. Again not sure if this will be here longterm though, I think this should be tied to the URL more but not sure how yet.

Reply:
Hmm, unfortunately my RxJs knowledge is kinda limited, but summoning @mckn, maybe he has an idea if there's a better or simpler approach. What [we are using in the extensions registry is this:](https://github.com/grafana/grafana/blob/bc7386e8155654907f5d7b06ab23795b0aa9542b/public/app/features/plugins/extensions/registry/Registry.ts#L75)
```ts
firstValueFrom(this._initialContext);
```

Reply:
Just a nit: what would be the general usecase for accessing the context - to get the current value or to subscribe to it? If it's more like getting the current value then it might make sense to switch the method names around:
```ts
get initialContextAsObservable() { }

get initialContext() {} 
```

Reply:
```
firstValueFrom(this._initialContext);
```
This would have the same problem wouldn't it? It returns a promise which needs to be handled in the hook so the first sync use of the hook does not have anything to return only the second one when the promise is resolved would return a value. I think usually this is not an issue having one additional render (maybe some perf?) but in the case of tracing app if I don't initialize some of the state the first time it does not react to second render because it has some uncontrolled components I assume which take just the initialValue prop.

Reply:
> what would be the general usecase for accessing the context - to get the current value or to subscribe to it?

Yeah I would think mostly just the initial value, because it should not change until some other app is rendered anyway. The naming you suggest makes sense it just feels weird right now next to the `activePluginId` accessors.